web-client/src/pages/admin/AdminSockDetailsPage.tsx

```diff
@@ -0,0 +1,244 @@
+
+export default function AdminSockDetailsPage() {
+  const { sockId } = useParams<{ sockId: string }>();
+  const numericSockId = Number(sockId);
+
+  const { data: sock, refetch, isLoading, isError } = useGetSockById(numericSockId);
+
+  const [isEditSockOpen, setEditSockOpen] = useState(false);
+  const [editVariant, setEditVariant] = useState<null | SockVariant>(null);
+  const [isAddVariantOpen, setAddVariantOpen] = useState(false);
+  const [currentVariants, setCurrentVariants] = useState<SockVariant[]>([]);
+
+  const updateSockMutation = useUpdateSockMutation();
+
+  useEffect(() => {
+    if (sock?.variants) {
+      setCurrentVariants(sock.variants);
+    }
+  }, [sock]);
+
+  const handleUpdateSock = async (updatedSock: UpdateSock) => {
+    try {
+      const payload = {
+        id: numericSockId,
+        ...updatedSock,
+      };
+      await updateSockMutation.mutateAsync(payload);
+      refetch();
+      setEditSockOpen(false);
+      toast.success("Sock updated successfully!");
+    } catch (error) {
+      console.error("Error updating sock:", error);
+      toast.error("Failed to update sock details.");
+    }
+  };
+
+  const handleAddVariant = async (newVariant: SockVariant) => {
+    try {
+      const updatedVariants: SockVariant[] = [
+        ...currentVariants,
+        {
+          ...newVariant,
+          size: newVariant.size as SockVariant["size"],
+        },
+      ];
+      setCurrentVariants(updatedVariants);
+
+      await updateSockMutation.mutateAsync({
+        id: numericSockId,
+        sock: {
+          name: sock!.name,
+          description: sock!.description,
+          previewImageUrl: sock!.previewImageUrl,
+        },
+        variants: updatedVariants,
+      });
+
+      refetch();
+      setAddVariantOpen(false);
+      toast.success("Variant added successfully!");
+    } catch (error) {
+      console.error("Error adding variant:", error);
+      toast.error("Failed to add variant.");
+    }
+  };
+
+  const handleEditVariant = async (updatedVariant: SockVariant) => {
+    try {
+      const updatedVariants: SockVariant[] = currentVariants.map((variant) =>
+        variant.id === updatedVariant.id ? updatedVariant : variant
+      );
+      setCurrentVariants(updatedVariants);
+
+      await updateSockMutation.mutateAsync({
+        id: numericSockId,
+        sock: {
+          name: sock!.name,
+          description: sock!.description,
+          previewImageUrl: sock!.previewImageUrl,
+        },
+        variants: updatedVariants,
+      });
+
+      refetch();
+      setEditVariant(null);
+      toast.success("Variant updated successfully!");
+    } catch (error) {
+      console.error("Error updating variant:", error);
+      toast.error("Failed to update variant.");
+    }
+  };
+
+  if (isLoading) return <div>Loading...</div>;
+  if (isError || !sock) return <div>Error fetching sock details.</div>;
+
+  const availableSizes = ["S", "M", "LG", "XL"].filter(
```

Comment:
Rename to nextSizesToSelect and import the existing `availableSizes` array from the `/inventory/model.ts`